File: etc/wrappers/wrapper.py
# Gets shelled out to by xiki to delegate call to a .py file.
# Just gets the args passed in and requires and invokes.
import sys, re

file = sys.argv[1]
path = sys.argv[2]

clazz = re.sub(".+/", "", file)
clazz = re.sub("\.py$", "", clazz)
clazz = clazz.capitalize()
execfile(file)

method = ".menu"

import re

if re.search("^\.", path):
  method = path.replace(r'/', '')

print(eval(clazz+"()"+method+"()"))

File: etc/wrappers/wrapper.rb
# Gets shelled out to by xiki to delegate call to a no-dependency .rb file.
# Just gets the args passed in and requires and invokes.

# TODO Don't hard-code path - use __FILE__?
# require "/projects/xiki/lib/xiki/ol.rb"

file = ARGV.shift
path = ARGV.shift

load file

clazz = file[/\w+/].gsub(/_([a-z]+)/) {"#{$1.capitalize}"}.sub(/(.)/) {$1.upcase}.gsub("_", "")
clazz = eval clazz

method = "menu"
method = path[/^\.(.+?)\/?$/, 1] if path =~ /^\./
method = method.to_sym

if clazz.respond_to? method
  puts clazz.send method
else
  cmethods = clazz.methods - Class.methods
  puts cmethods.sort.map{|o| "+ .#{o}/"}
end

